internal/api/token.go

```diff
@@ -199,17 +199,17 @@ func (a *API) ResourceOwnerPasswordGrant(ctx context.Context, w http.ResponseWri
 					return err
 				}
 			}
-			return oauthError("invalid_grant", InvalidLoginMessage)
+			return badRequestError(ErrorCodeInvalidLoginCredentials, InvalidLoginMessage)
 		}
 	}
 	if !isValidPassword {
-		return oauthError("invalid_grant", InvalidLoginMessage)
+		return badRequestError(ErrorCodeInvalidLoginCredentials, InvalidLoginMessage)
```

Comment:
Not too sure how to decide between `badRequest` Error and unprocessable entity here.

Reply:
badRequest for backward compatibility because that's what oauthError uses - i prefer unprocessible entity too

Reply:
makes sense -let's keep it then